sys/vane/clay.hoon

```diff
@@ -3576,19 +3703,33 @@
     |=  {pot/term bem/beam}
     [u.hez.ruf %give %ogre pot]
   ::
-      $warp
+      $perm
+    =^  mos  ruf
+      ::TODO  after new boot system, just use our from global.
+      =+  den=((de now hen ruf) [. .]:our.q.hic des.q.hic)
+      abet:(perm:den pax.q.hic rit.q.hic)
+    [mos ..^$]
+  ::
+      ?($warp $werp)
+    =^  for  q.hic
+      ?:  ?=($warp -.q.hic)
+        [~ q.hic]
+      :_  [%warp q.q.hic r.q.hic]
+      ?:  =(p.q.hic p.q.q.hic)  ~&([%huh-this-west-may-be-weird p.q.hic] ~)
+      `p.q.hic
+    ?>  ?=($warp -.q.hic)
     =^  mos  ruf
       =+  den=((de now hen ruf) p.q.hic p.q.q.hic)
-      ::  =-  ~?  ?=([~ %sing %w *] q.q.q.hic)
+      ::  =-  ~?  ?=([~ %sing %w *] q.r.q.hic)
       ::        :*  %someones-warping
-      ::            rav=u.q.q.q.hic
+      ::            rav=u.q.r.q.hic
```

Comment:
There's no way I can reasonably review `u.q.r.q.hic`. I realize this was already the style in this part of the code, but it's not fit for human consumption.

Reply:
This particular comment can probably just get removed, but I agree that I should probably add a few `=*` here.